### PR TITLE
google-cloud-logging-logback: Allow user to specify custom LoggingOptions

### LINKED ISSUE #3215: Injecting google cloud credentials instead GOOGLE_APPLICATION_CREDENTIALS environment
Please provide a way to inject google credentials information for the stackdriver logger by means of public method/property instead of placing a private key file on machine and refer it from the GOOGLE_APPLICATION_CREDENTIALS environment variable.

Thanks.

### PR DESCRIPTION
**Pull request by olavloite**
Allow users to subclass com.google.cloud.logging.logback.LoggingAppender to specify custom LoggingOptions, such as custom credentials.
Fixes #3215

### CODE REVIEW

**Comment by ajaaym** [comment]
WDYT about adding setter method for credential file so user can configure from xml rather than writing code to subclass?

**Comment by olavloite** [comment]
> WDYT about adding setter method for credential file so user can configure from xml rather than writing code to subclass?

Yes, you are absolutely right. I'll change it to support that.

File: google-cloud-clients/google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/logback/LoggingAppender.java [hunk 1]
```diff
@@ -212,13 +230,32 @@ Logging getLogging() {
...
-          logging = LoggingOptions.getDefaultInstance().getService();
+          logging = createLoggingOptions().getService();
         }
       }
     }
     return logging;
   }
 
+  /** Creates the {@link LoggingOptions} to use for this {@link LoggingAppender}. */
+  LoggingOptions createLoggingOptions() {
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/logback/LoggingAppender.java]: Can we do a similar approach as LoggingHandler (saving loggingOptions as a variable)?

Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/logback/LoggingAppender.java]: Changed to store the LoggingOptions as a variable.

File: google-cloud-clients/google-cloud-contrib/google-cloud-logging-logback/README.md [hunk 2]
```diff
@@ -88,6 +88,23 @@ Authentication
 See the [Authentication](https://github.com/googleapis/google-cloud-java#authentication)
 section in the base directory's README.
 
+You can also specify custom credentials and other options by creating a subclass of `com.google.cloud.logging.logback.LoggingAppender` and override the method `createLoggingOptions()`. Your logback.xml configuration file must reference your subclass instead of `com.google.cloud.logging.logback.LoggingAppender`.
+
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-logging-logback/README.md]: do we still need these instructions for overriding createLoggingOptions?

Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-logging-logback/README.md]: You're right, I changed these instructions into using the new custom property fileCredentials.

**Pull request merged**